crates/apub_receive/src/activities/post/undo_remove.rs

```diff
@@ -0,0 +1,58 @@
+use crate::activities::{
+  post::{remove::RemovePost, send_websocket_message},
+  verify_mod_action,
+};
+use activitystreams::activity::kind::UndoType;
+use lemmy_api_common::blocking;
+use lemmy_apub::{check_is_apub_id_valid, fetcher::objects::get_or_fetch_and_insert_post};
+use lemmy_apub_lib::{verify_domains_match, ActivityCommonFields, ActivityHandlerNew, PublicUrl};
+use lemmy_db_queries::source::post::Post_;
+use lemmy_db_schema::source::post::Post;
+use lemmy_utils::LemmyError;
+use lemmy_websocket::{LemmyContext, UserOperationCrud};
+use url::Url;
+
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UndoRemovePost {
+  to: PublicUrl,
+  object: RemovePost,
+  cc: [Url; 1],
+  #[serde(rename = "type")]
+  kind: UndoType,
+  #[serde(flatten)]
+  common: ActivityCommonFields,
+}
+
+#[async_trait::async_trait(?Send)]
+impl ActivityHandlerNew for UndoRemovePost {
+  async fn verify(
+    &self,
+    context: &LemmyContext,
+    request_counter: &mut i32,
+  ) -> Result<(), LemmyError> {
+    verify_domains_match(&self.common.actor, self.common.id_unchecked())?;
+    check_is_apub_id_valid(&self.common.actor, false)?;
+    verify_mod_action(self.common.actor.clone(), self.cc[0].clone(), context).await?;
+    self.object.verify(context, request_counter).await
+  }
```

Comment:
This seems so common, maybe it could be a default function on the trait?

Reply:
I need to rework all the verify() functions, so I will see then how to extract common code. A trait wont work because it cant access struct fields, it would require a macro (or passing the individual values directly into a helper function).

Reply:
You can add functions to a trait that refer to a structs fields, then when you implement it, you refer to them:

```rust
trait ActorCommon {
  fn actor() -> ...
```

Reply:
I'm not sure how that would help to simplify anything. And anyway this PR is already huge, so I will finish it this way, and we can think about further changes later.